File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.py
def solution(n):
    a = ''
    result = ''
    
    #10 진법 > 3진법
    while n > 0:
        a = str(n % 3) + a
        n //= 3
        
    #3진법 뒤집기
    reverse_n = a[::-1]
    
    #3진법 > 10진법
    result = int(reverse_n, 3)
        
    return result

File: 프로그래머스/1/68935. 3진법 뒤집기/3진법 뒤집기.java
class Solution {
    public int solution(int n) {
        // 1. 10진법 -> 3진법으로 변환 및 뒤집기
        String ternary = "";
        while (n > 0) {
            ternary = (n % 3) + ternary;
            n /= 3;
        }
        
        // 2. 뒤집힌 3진법 -> 10진법으로 변환
        int answer = 0;
        int power = 1;
        for (int i = 0; i < ternary.length(); i++) {
            answer += (ternary.charAt(i) - '0') * power;
            power *= 3;
        }
        
        return answer;
    }
}
